### PR TITLE
libpopcnt: add recipe

### PR DESCRIPTION
**Pull request by toge**
### Summary
Changes to recipe:  **libpopcnt/3.1**

#### Motivation
Fast C/C++ bit population count library.

#### Details
https://github.com/kimwalisch/libpopcnt

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Review by jcar87 — CHANGES_REQUESTED** [review]

File: recipes/libpopcnt/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def package_id(self):
+        self.info.clear()
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
```
[the remark this answers is not in the document]
Comment (toge) [reply recipes/libpopcnt/all/conanfile.py:27]: Fixed.

File: recipes/libpopcnt/all/test_package/test_package.c [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#include "libpopcnt.h"
+
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+
+int main() {
+  int i = 0;
+  size_t size = 70000;
```
Comment (AbrilRBS) [inline recipes/libpopcnt/all/test_package/test_package.c:9]:
```suggestion
  size_t size = 42;
```

**Review by AbrilRBS — DISMISSED** [review]
Thanks!

**Review by AbrilRBS — APPROVED** [review]
Thanks!

**Review by jcar87 — APPROVED** [review]

**Review by AbrilRBS — APPROVED** [review]

**Pull request merged**